### PR TITLE
network expansion 

### PR DESCRIPTION
**Pull request by walmat**
More info https://rainbowhaus.slack.com/archives/C0468CDBE75/p1727380431942769?thread_ts=1727377244.944789&cid=C0468CDBE75

Fixes APP-1886

### CODE REVIEW

File: src/__swaps__/screens/Swap/components/AnimatedChainImage.android.tsx [hunk 1]
```diff
@@ -12,33 +12,35 @@ const ZoraBadge = require('@/assets/badges/zora.png');
...
 const BlastBadge = require('@/assets/badges/blast.png');
 const DegenBadge = require('@/assets/badges/degen.png');
+const ApechainBadge = require('@/assets/badges/apechainBadge.png');
 
 import { ChainId } from '@/chains/types';
 import { globalColors } from '@/design-system';
 import { PIXEL_RATIO } from '@/utils/deviceUtils';
 import { useSwapsStore } from '@/state/swaps/swapsStore';
 
 const networkBadges = {
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/AnimatedChainImage.android.tsx:22]: alphabetized these.

File: src/__swaps__/screens/Swap/components/AnimatedChainImage.ios.tsx [hunk 2]
```diff
@@ -22,26 +23,27 @@ import { useSwapContext } from '../providers/swap-provider';
 import { BLANK_BASE64_PIXEL } from '@/components/DappBrowser/constants';
 
 const networkBadges = {
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/AnimatedChainImage.ios.tsx:25]: alphabetized these.

File: src/__swaps__/screens/Swap/resources/search/search.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ import { parseTokenSearch } from './utils';
 const ALL_VERIFIED_TOKENS_PARAM = '/?list=verifiedAssets';
 
 const tokenSearchHttp = new RainbowFetchClient({
-  baseURL: 'https://token-search.rainbow.me/v2',
+  baseURL: 'https://token-search-v2.rainbowdotme.workers.dev/v2',
```
Comment (walmat) [inline src/__swaps__/screens/Swap/resources/search/search.ts:15]:
```suggestion
  baseURL: 'https://token-search.rainbow.me/v2',
```

File: src/handlers/tokenSearch.ts [hunk 4]
```diff
@@ -5,15 +5,14 @@ import { TokenSearchThreshold, TokenSearchTokenListId, TokenSearchUniswapAssetKe
...
-import ethereumUtils from '@/utils/ethereumUtils';
 import { chainsName } from '@/chains';
 
 type TokenSearchApiResponse = {
   data: TokenSearchToken[];
 };
 
 const tokenSearchApi = new RainbowFetchClient({
-  baseURL: 'https://token-search.rainbow.me/v2',
+  baseURL: 'https://token-search-v2.rainbowdotme.workers.dev/v2',
```
Comment (walmat) [inline src/handlers/tokenSearch.ts:15]:
```suggestion
  baseURL: 'https://token-search.rainbow.me/v2',
```

File: src/handlers/tokenSearch.ts [hunk 5]
```diff
@@ -105,6 +104,7 @@ export const tokenSearch = async (searchParams: {
       };
     });
   } catch (e: any) {
+    console.log(e);
```
Comment (walmat) [inline src/handlers/tokenSearch.ts:107]:
```suggestion
```

File: src/model/remoteConfig.ts [hunk 6]
```diff
@@ -102,16 +102,17 @@ export const DEFAULT_CONFIG: RainbowConfig = {
   data_endpoint: DATA_ENDPOINT || 'wss://api-v4.zerion.io',
   data_origin: DATA_ORIGIN,
```
Comment (walmat) [inline src/model/remoteConfig.ts:103]: alphabetizes these.

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/APP-1893/12434a6e4c690e4455c3d52206b8277c951779a5.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/APP-1893/12434a6e4c690e4455c3d52206b8277c951779a5.ipa) for 12434a6e4c690e4455c3d52206b8277c951779a5

**Review by christianbaroni — APPROVED** [review]

File: src/references/index.ts [hunk 7]
```diff
@@ -161,6 +162,7 @@ export const NATIVE_ASSETS_PER_CHAIN: Record<ChainId, AddressOrEth> = {
   [ChainId.blastSepolia]: AddressZero,
   [ChainId.polygonAmoy]: AddressZero,
   [ChainId.degen]: AddressZero,
+  [ChainId.apechain]: AddressZero,
```
Comment (estebanmino) [inline src/references/index.ts:165]: i think NATIVE_ASSETS_PER_CHAIN can be removed now and use `chainsNativeAsset` 

import { chainsNativeAsset } from '@/chains';

File: src/state/appSessions/index.ts [hunk 8]
```diff
@@ -4,6 +4,7 @@ import { Network, ChainId } from '@/chains/types';
 import { createRainbowStore } from '../internal/createRainbowStore';
 
 const chainsIdByNetwork: Record<Network, ChainId> = {
```
Comment (estebanmino) [inline src/state/appSessions/index.ts:6]: `chainsIdByNetwork` is only used in an old state migration for `appSession` so we don't need to keep adding chains

File: src/styles/colors.ts [hunk 9]
```diff
@@ -199,6 +199,7 @@ const getColorsByTheme = (darkMode?: boolean) => {
     [ChainId.avalanche]: '#E84142',
     [ChainId.degen]: '#A36EFD',
     [ChainId.blast]: '#25292E',
+    [ChainId.apechain]: '#0054FA',
```
Comment (estebanmino) [inline src/styles/colors.ts:202]: ok i missed this one on my PRs, is it possible to use native assets color? or maybe we're missing this color from the backend data

**Review by estebanmino — APPROVED** [review]
lgtm

maybe we could add chain images and colors from backend too? I didn't add them in my PRs

**Comment by walmat** [comment]
> lgtm
> 
> maybe we could add chain images and colors from backend too? I didn't add them in my PRs

For sure! I think both would be a great addition to the network expansion work. iirc we haven't done images yet because of how shadows work in the app but I may be wrong (cc @christianbaroni)

File: src/styles/colors.ts (hunk 9 above)
Comment (walmat) [reply src/styles/colors.ts:202]: Looked into this, they're different from native asset colors so I dunno. @christianbaroni any idea on this?

**Pull request merged**